src/app/containers/card-deck/card-deck.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, OnInit, Renderer2 } from '@angular/core';
+import { MonsterCard } from '../../models/card.model';
+
+@Component({
+    selector: 'card-deck',
+    styleUrls: ['card-deck.component.scss'],
+    template: `
+        <div *ngFor="let c of monsterCards" class="card-container">
+            <monster-card (click)="cardClick(c)"
+                [ngClass]="{
+                    'flipped flip-to-front': c.facing === 'up',
+                    'send-to-back': c.facing === 'down',
+                    'shuffle': this.shuffleDeck === true
+                }"
+                [title]="c.title"
+                [order]="c.order"
+                [icon]="c.icon"
+                [body]="c.body"
+                [facing]="c.facing"
+            ></monster-card>
+        </div>
+    `,
+})
+
+export class CardDeckComponent implements OnInit{
+
+    monsterCards: MonsterCard[] = [];
+
+    shuffleDeck: boolean = false;
+
+    constructor() { }
+
+    ngOnInit() {
+
+        // generate some goofy test cards
+        for (let i = 0; i < 10; i++) {
+            let ord = Math.floor(Math.random() * 100);
+            this.monsterCards.push(new MonsterCard('Flame Demon - 1', ord, 'icon ' + i.toString(), 'Some card description ' + i.toString()))
+        }
+
+        this.sortInOrder(this.monsterCards, 'order');
+
+        // add in a test shuffle card
+        this.monsterCards[5].shuffle = true;
+
+        // console.log(this.monsterCards);
+    }
+
+    private cardClick(card) {
+        if (card.shuffle === true) {
```

Comment:
The deck should not shuffle if a card with the shuffle flag shows up. The deck will be shuffled at the end of the round. On card-clicks, you should just be checking if the revealed card is a 'shuffle-trigger' card and update a local variable to be referenced later.

Reply:
👍